velox/exec/fuzzer/PrestoQueryRunnerIntermediateTypeTransforms.cpp

```diff
@@ -0,0 +1,444 @@
+                    std::make_shared<core::FieldAccessExpr>("v", "v"),
+                    "v"))},
+        columnAlias);
+  }
+
+  return expr;
+}
+
+// Converts an RowVector so that any intermediate only types in the children
+// are transformed.
+VectorPtr RowTransform::transform(
+    const VectorPtr& vector,
+    const SelectivityVector& rows) const {
+  VELOX_CHECK(vector->type()->isRow());
+  DecodedVector decoded(*vector);
+  const auto* base = decoded.base()->as<RowVector>();
+
+  SelectivityVector childRows(base->size(), false);
+  rows.applyToSelected([&](vector_size_t row) {
+    if (!decoded.isNullAt(row)) {
+      childRows.setValid(decoded.index(row), true);
+    }
+  });
+  childRows.updateBounds();
+
+  std::vector<VectorPtr> children;
+  std::vector<TypePtr> childrenTypes;
+  std::vector<std::string> childrenNames = base->type()->asRow().names();
+  for (const auto& child : base->children()) {
+    if (isIntermediateOnlyType(child->type())) {
+      children.push_back(transformIntermediateOnlyType(child, childRows));
+      childrenTypes.push_back(children.back()->type());
+    } else {
+      children.push_back(child);
+      childrenTypes.push_back(child->type());
+    }
+  }
+
+  VectorPtr row = std::make_shared<RowVector>(
+      base->pool(),
+      ROW(std::move(childrenNames), std::move(childrenTypes)),
+      base->nulls(),
+      base->size(),
+      std::move(children));
+
+  if (!decoded.isIdentityMapping()) {
+    row = decoded.wrap(row, *vector, vector->size());
+  }
+
+  return row;
+}
+
+// Applies transforms to the children of a row to convert input types to
+// intermediate only types where necessary, and reconstructs the row via
+// row_constructor.
+core::ExprPtr RowTransform::projectionExpr(
+    const TypePtr& type,
+    const core::ExprPtr& inputExpr,
+    const std::string& columnAlias) const {
+  VELOX_CHECK(type->isRow());
+  const auto& rowType = type->asRow();
+
+  std::vector<core::ExprPtr> children;
+  for (int i = 0; i < rowType.size(); i++) {
+    if (isIntermediateOnlyType(rowType.childAt(i))) {
+      children.push_back(getIntermediateOnlyTypeProjectionExpr(
+          rowType.childAt(i),
+          std::make_shared<core::FieldAccessExpr>(
+              rowType.nameOf(i),
+              rowType.nameOf(i),
+              std::vector<core::ExprPtr>{inputExpr}),
+          rowType.nameOf(i)));
+    } else {
+      children.push_back(std::make_shared<core::FieldAccessExpr>(
+          rowType.nameOf(i),
+          rowType.nameOf(i),
+          std::vector<core::ExprPtr>{inputExpr}));
+    }
+  }
+
+  return std::make_shared<core::CallExpr>(
+      "switch",
+      std::vector<core::ExprPtr>{
+          std::make_shared<core::CallExpr>(
+              "is_null", std::vector<core::ExprPtr>{inputExpr}, std::nullopt),
+          std::make_shared<core::ConstantExpr>(
+              type, variant::null(TypeKind::ROW), std::nullopt),
+          std::make_shared<core::CastExpr>(
+              type,
```

Comment:
So this `type` argument is the RowType that contains intermediate types, while `inputExpr` is of a RowType that contains input types? Could we add comments in the header file to document these arguments?

Reply:
I can just get rid of type from the public interface, we only need it for converting complex types that contain intermediate types, which only happens within this cpp file.